### PR TITLE
TreeOptimizer remove unnecessary dict get special case

### PR DESCRIPTION
**Pull request by kitaisreal**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Interpreters/TreeOptimizer.cpp [hunk 1]
```diff
@@ -115,34 +97,7 @@ void optimizeGroupBy(ASTSelectQuery * select_query, const NameSet & source_colum
...
-
-                const auto & dict_ptr = context.getExternalDictionariesLoader().getDictionary(dict_name, context);
-                if (!dict_ptr->isInjective(attr_name))
-                {
-                    ++i;
-                    continue;
-                }
-            }
-            else if (!function_factory.get(function->name, context)->isInjective({}))
+            if (!function_factory.get(function->name, context)->isInjective({}))
```
Comment (kitaisreal) [inline src/Interpreters/TreeOptimizer.cpp:100]: DictGet function already implement isInjective same way as code before.

Comment (alexey-milovidov) [reply src/Interpreters/TreeOptimizer.cpp:100]: How does it check for function arguments (dictionary and attribute name)?

Comment (kitaisreal) [reply src/Interpreters/TreeOptimizer.cpp:100]: It does the same procedure as before. 
```
const auto & dict_ptr = context.getExternalDictionariesLoader().getDictionary(dict_name, context);
!dict_ptr->isInjective(attr_name)
```

**Comment by alexey-milovidov** [comment]
We figured out that this change is wrong and we need first to create a test to prove that it is wrong.

**Pull request closed without merging**